### PR TITLE
fix(platformstatus): fix multiflag usage and improve messages formatting

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description

This PR throw an exception while using the same flag more than once and improve the messages formatting.

**Fixes** # MON-6214

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 19.10.x
- [ ] 20.04.x
- [x] 20.10.x
- [x] 21.04.x (master)

<h2> How this pull request can be tested ? </h2>

Execute the script with only one network interface, you shouldn't be asked to choose the correct network. Otherwise fill the Template with the CURRENT_NODE_ADDRESS option.

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: bin/registerServerTopology-Func.php [hunk 1]
```diff
@@ -316,8 +315,7 @@ function setConfigOptionsFromTemplate(array $options, string $helpMessage): arra
 
     if (!$configOptions['CURRENT_NODE_TYPE']) {
         throw new \InvalidArgumentException(
-            "CURRENT_NODE_TYPE must be one of those value"
-            . PHP_EOL . "Poller, Remote, MAP, MBI" . PHP_EOL
+            "CURRENT_NODE_TYPE must be one of those value: Poller, Remote, MAP, MBI"
```
Comment (kduret) [inline bin/registerServerTopology-Func.php:318]:
```suggestion
            "CURRENT_NODE_TYPE must be one of those values: Poller, Remote, MAP, MBI"
```

File: bin/registerServerTopology.php [hunk 2]
```diff
@@ -76,7 +76,18 @@
              - PROXY_PASSWORD           <optional> string
 
 EOD;
-
+try {
+    foreach ($opt as $key => $option) {
+        if (is_array($option)) {
+            throw new \InvalidArgumentException("
+                each flag must be declared only once : -$key as been declared twice or more
```
Comment (kduret) [inline bin/registerServerTopology.php:83]:
```suggestion
                each flag must be declared only once: --$key as been declared twice or more
```

File: bin/registerServerTopology.php [hunk 3]
```diff
@@ -188,25 +198,29 @@
  */
 $foundIps = explode(" ", trim(shell_exec("hostname -I")));
 $foundIps = array_combine(range(1, count($foundIps)), array_values($foundIps));
+if (count($foundIps) > 1 && !isset($configOptions['CURRENT_NODE_ADDRESS'])) {
+    $goodIp = false;
 
-$goodIp = false;
+    $ipSelection = 'Found IP on CURRENT NODE:' . PHP_EOL;
+    foreach ($foundIps as $key => $ip) {
+            $ipSelection .= "   [$key]: $ip" . PHP_EOL;
```
Comment (kduret) [inline bin/registerServerTopology.php:206]:
```suggestion
        $ipSelection .= "   [$key]: $ip" . PHP_EOL;
```

File: bin/registerServerTopology.php (hunk 2 above)
Comment (sc979) [inline bin/registerServerTopology.php:83]:
```suggestion
                each flag must be declared only once : $key has been declared more than once
```
no dash should be displayed has you don't distinguish single and double dashes : -h vs --root

File: bin/registerServerTopology.php [hunk 4]
```diff
@@ -112,8 +123,7 @@
 
         if (!$configOptions['CURRENT_NODE_TYPE']) {
             throw new \InvalidArgumentException(
-                "-t must be one of those value"
-                    . PHP_EOL . "Poller, Remote, MAP, MBI" . PHP_EOL
+                "-t must be one of those value: Poller, Remote, MAP, MBI"
```
Comment (sc979) [inline bin/registerServerTopology.php:126]:
```suggestion
                "-t must be one of those values: Poller, Remote, MAP, MBI"
```

File: bin/registerServerTopology.php [hunk 5]
```diff
@@ -188,25 +198,29 @@
...
+    $ipSelection = 'Found IP on CURRENT NODE:' . PHP_EOL;
+    foreach ($foundIps as $key => $ip) {
+            $ipSelection .= "   [$key]: $ip" . PHP_EOL;
+    }
 
-$ipSelection = 'Found IP on CURRENT NODE:' . PHP_EOL;
-foreach ($foundIps as $key => $ip) {
-        $ipSelection .= "   [$key]: $ip" . PHP_EOL;
-}
+    while (!$goodIp) {
```
Comment (sc979) [inline bin/registerServerTopology.php:209]:
```suggestion
    while ($goodIp === false) {
```

File: bin/registerServerTopology.php [hunk 6]
```diff
@@ -188,25 +198,29 @@
...
-        $goodIp = true;
+        if (!array_key_exists($ipChoice, $foundIps)) {
+            echo 'Bad IP Choice' . PHP_EOL;
+        } else {
+            $goodIp = true;
+        }
     }
+    $serverIp = $foundIps[$ipChoice];
+} else {
+    $serverIp = $foundIps[1];
```
Comment (sc979) [inline bin/registerServerTopology.php:222]: not $foundIps[0]. What should contains $foundIps[0] ?

File: bin/registerServerTopology.php [hunk 7]
```diff
@@ -188,25 +198,29 @@
  */
 $foundIps = explode(" ", trim(shell_exec("hostname -I")));
 $foundIps = array_combine(range(1, count($foundIps)), array_values($foundIps));
+if (count($foundIps) > 1 && !isset($configOptions['CURRENT_NODE_ADDRESS'])) {
+    $goodIp = false;
```
Comment (sc979) [inline bin/registerServerTopology.php:202]: maybe $validateIp is more clear than $goodIp

**Review by sc979 — CHANGES_REQUESTED** [review]

File: bin/registerServerTopology.php (hunk 2 above)
Comment (sc979) [reply bin/registerServerTopology.php:83]: -h : single dash
--root : double dash
-> https://github.com/centreon/centreon/pull/9175#discussion_r515804138

File: bin/registerServerTopology.php (hunk 7 above)
Comment (adr-mo) [reply bin/registerServerTopology.php:202]: @sc979 validate is a verb I would say as it is a boolean `$validIP`

File: bin/registerServerTopology.php (hunk 6 above)
Comment (jeremyjaouen) [reply bin/registerServerTopology.php:222]: nothing, the array is re indexed from 1

**Review by sc979 — APPROVED** [review]

**Review by kduret — APPROVED** [review]

**Pull request merged**